src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -398,29 +447,34 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
             return null;
         }
 
-        private SeparatedSyntaxList<T> PermuteDeclaration<T>(SeparatedSyntaxList<T> list, SignatureChange updatedSignature) where T : SyntaxNode
+        private static ParameterSyntax CreateNewParameterSyntax(AddedParameter addedParameter)
+            => CreateNewParameterSyntax(addedParameter, skipParameterType: false);
+
+        private static ParameterSyntax CreateNewParameterSyntax(AddedParameter addedParameter, bool skipParameterType)
+            => SyntaxFactory.Parameter(
+                attributeLists: SyntaxFactory.List<AttributeListSyntax>(),
+                modifiers: SyntaxFactory.TokenList(),
+                type: skipParameterType
+                    ? default
+                    : addedParameter.Type.GenerateTypeSyntax(allowVar: false)
+                        .WithTrailingTrivia(SyntaxFactory.ElasticSpace),
+                SyntaxFactory.Identifier(addedParameter.ParameterName),
```

Comment:
Does this need to do escaping?

Reply:
If you try to name the parameter `for` (for example), the Add Parameter dialog rejects it (with an inappropriate message, but that's a different issue), so it would already have to be escaped by the time it gets here.